Index: qpsmtpd/plugins/auth/auth_ldap_bind
===================================================================
--- qpsmtpd.orig/plugins/auth/auth_ldap_bind
+++ qpsmtpd/plugins/auth/auth_ldap_bind
@@ -81,7 +81,7 @@ Please see the LICENSE file included wit
 use strict;
 use warnings;
 
-use Net::LDAP qw(:all);
+use Net::LDAP;
 use Qpsmtpd::Constants;
 
 sub register {
@@ -173,7 +173,7 @@ sub authldap {
     $ldh->unbind if $ldh;
 
     # bind against directory as user with password supplied
-    if (!$mesg->count || $lduserdn = $mesg->entry->dn) {
+    if (!$mesg->count or $lduserdn = $mesg->entry->dn) {
         $self->log(LOGALERT, "fail: user not found");
         return DECLINED, "authldap - wrong username or password";
     }
