Index: qpsmtpd/plugins/check_badmailfrom
===================================================================
--- /dev/null
+++ qpsmtpd/plugins/check_badmailfrom
@@ -0,0 +1,14 @@
+# Backwards-compatibility wrapper; qpsmtpd 0.94 renamed the check_* plugins
+# from the upstream distribution to remove the check_ prefix.  This plugin
+# exists to allow older, un-migrated configurations to work properly with
+# the 0.94 release.
+
+my $whoami = 'badmailfrom';
+
+sub init {
+  my $self = shift;
+  $self->log(LOGINFO,
+             "The check_$whoami plugin is deprecated; see" .
+             "/usr/share/doc/qpsmtpd/README.check_plugins for details.");
+  $self->isa_plugin($whoami);
+}
Index: qpsmtpd/plugins/check_badrcptto
===================================================================
--- /dev/null
+++ qpsmtpd/plugins/check_badrcptto
@@ -0,0 +1,14 @@
+# Backwards-compatibility wrapper; qpsmtpd 0.94 renamed the check_* plugins
+# from the upstream distribution to remove the check_ prefix.  This plugin
+# exists to allow older, un-migrated configurations to work properly with
+# the 0.94 release.
+
+my $whoami = 'badrcptto';
+
+sub init {
+  my $self = shift;
+  $self->log(LOGINFO,
+             "The check_$whoami plugin is deprecated; see" .
+             "/usr/share/doc/qpsmtpd/README.check_plugins for details.");
+  $self->isa_plugin($whoami);
+}
Index: qpsmtpd/plugins/check_earlytalker
===================================================================
--- /dev/null
+++ qpsmtpd/plugins/check_earlytalker
@@ -0,0 +1,14 @@
+# Backwards-compatibility wrapper; qpsmtpd 0.94 renamed the check_* plugins
+# from the upstream distribution to remove the check_ prefix.  This plugin
+# exists to allow older, un-migrated configurations to work properly with
+# the 0.94 release.
+
+my $whoami = 'earlytalker';
+
+sub init {
+  my $self = shift;
+  $self->log(LOGINFO,
+             "The check_$whoami plugin is deprecated; see" .
+             "/usr/share/doc/qpsmtpd/README.check_plugins for details.");
+  $self->isa_plugin($whoami);
+}
Index: qpsmtpd/plugins/check_loop
===================================================================
--- /dev/null
+++ qpsmtpd/plugins/check_loop
@@ -0,0 +1,14 @@
+# Backwards-compatibility wrapper; qpsmtpd 0.94 renamed the check_* plugins
+# from the upstream distribution to remove the check_ prefix.  This plugin
+# exists to allow older, un-migrated configurations to work properly with
+# the 0.94 release.
+
+my $whoami = 'loop';
+
+sub init {
+  my $self = shift;
+  $self->log(LOGINFO,
+             "The check_$whoami plugin is deprecated; see" .
+             "/usr/share/doc/qpsmtpd/README.check_plugins for details.");
+  $self->isa_plugin($whoami);
+}
Index: qpsmtpd/plugins/check_relay
===================================================================
--- /dev/null
+++ qpsmtpd/plugins/check_relay
@@ -0,0 +1,14 @@
+# Backwards-compatibility wrapper; qpsmtpd 0.94 renamed the check_* plugins
+# from the upstream distribution to remove the check_ prefix.  This plugin
+# exists to allow older, un-migrated configurations to work properly with
+# the 0.94 release.
+
+my $whoami = 'relay';
+
+sub init {
+  my $self = shift;
+  $self->log(LOGINFO,
+             "The check_$whoami plugin is deprecated; see" .
+             "/usr/share/doc/qpsmtpd/README.check_plugins for details.");
+  $self->isa_plugin($whoami);
+}
Index: qpsmtpd/plugins/check_spamhelo
===================================================================
--- /dev/null
+++ qpsmtpd/plugins/check_spamhelo
@@ -0,0 +1,12 @@
+# Backwards-compatibility wrapper; qpsmtpd 0.94 renamed the check_* plugins
+# from the upstream distribution to remove the check_ prefix.  This plugin
+# exists to allow older, un-migrated configurations to work properly with
+# the 0.94 release.
+
+sub init {
+  my $self = shift;
+  $self->log(LOGINFO,
+             "The check_spamhelo plugin is deprecated; see" .
+             "/usr/share/doc/qpsmtpd/README.check_plugins for details.");
+  $self->isa_plugin('helo');
+}
